source/initial_temperature/anelasticity_vs_to_temperature.cc

```diff
@@ -94,43 +90,29 @@ namespace aspect
         {
           // convert absolute Vs into temperature
           // check if using Yamauchi & Takei 2016 parameterization
-          if (use_yamauchi_takei == true)
-            {
-              // specify Brent algorithm parameters
-              const double a=273;
-              const double b=3273;
-              typedef std::pair<double, double> Result;
-              // create fVs function to use in Brent minimization and calculate temperature
-              auto bfunc = [ &,this] (double x)
-              {
-                return fVs(x, depth, absolute_Vs, mu0, dmudT,dmudP,
-                           viscosity_prefactor,activation_energy,activation_volume,solidus_gradient,use_original_model);
-              };
-              // determine maximum Vs
-              double maximum_Vs;
-              int fail;
-              maximum_Vs=yamauchi_takei_Vs(273.,depth,mu0,dmudT,dmudP,viscosity_prefactor,activation_energy,activation_volume
-                                           ,solidus_gradient,use_original_model);
+          // specify Brent algorithm parameters
+          const double a=273;
+          const double b=3273;
+          // create fVs function to use in Brent minimization and calculate temperature
+          auto bfunc = [ &,this] (const double x)
+          {
+            return fVs(x, depth, absolute_Vs, mu0, dmudT,dmudP,
+                       viscosity_prefactor,activation_energy,activation_volume,solidus_gradient,use_original_model);
+          };
+          // determine maximum Vs
+          const double maximum_Vs=yamauchi_takei_Vs(273.,depth,mu0,dmudT,dmudP,viscosity_prefactor,activation_energy,activation_volume
+                                                    ,solidus_gradient,use_original_model);
 
-              // set number of fails to zero
-              fail=0;
-              // where absolute Vs exceeds maximum Vs, set temperature to 273 K
-              if (absolute_Vs>maximum_Vs)
-                {
-                  temperature=273.;
-                  fail=fail+1;
-                  std::cout << "Vs too fast for sensible temperature for " << fail << " points!" << std::endl;
-                }
-              else
-                {
-                  Result r1 = boost::math::tools::brent_find_minima(bfunc,a,b,16);
-                  temperature=r1.first;
-                }
+          // where absolute Vs exceeds maximum Vs, set temperature to 273 K
+          if (absolute_Vs>maximum_Vs)
+            {
+              AssertThrow(absolute_Vs > maximum_Vs,
```

Comment:
Hmm I am still not quite sure what you want to do here. If the absolute Vs is bigger than the maximum Vs, do you (i) want the model to crash (because the computation makes no sense anymore), or (ii) use 273 because that is a reasonable temperature for this case? If (i) then change then remove the `if-else` altogether and change the  AssertThrow to the following `AssertThrow(absolute_Vs < maximum_Vs, ExcMessage...)`. If (ii) then remove the AssertThrow. We currently have no nice way of writing warnings so in that case the plugin should simply continue.

Reply:
The second case is preferable. Occasionally you get a stray Vs value that is anomalously fast and its better that this doesn't crash the model.